internal/circle/client.go

```diff
@@ -264,6 +270,33 @@ func parseRetryAfter(retryAfter string) time.Duration {
 	return time.Duration(seconds) * time.Second
 }
 
+func (client *Client) recordCircleAPIMetrics(ctx context.Context, method, endpoint string, startTime time.Time, resp *http.Response, reqErr error) {
+	t, err := tenant.GetTenantFromContext(ctx)
+	if err != nil {
+		log.Ctx(ctx).Errorf("getting tenant from context: %v", err)
+		return
+	}
+
+	duration := time.Since(startTime)
+	status, statusCode := monitor.ParseHTTPResponseStatus(resp, reqErr)
+
+	labels := monitor.CircleLabels{
+		Method:     method,
+		Endpoint:   endpoint,
+		Status:     status,
+		StatusCode: statusCode,
+		TenantName: t.Name,
+	}.ToMap()
+
+	if monitorErr := client.monitorService.MonitorHistogram(duration.Seconds(), monitor.CircleAPIRequestDurationTag, labels); monitorErr != nil {
+		log.Ctx(ctx).Errorf("monitoring histogram: %v", err)
+	}
+
+	if monitorErr := client.monitorService.MonitorCounters(monitor.CircleAPIRequestsTotalTag, labels); monitorErr != nil {
+		log.Ctx(ctx).Errorf("monitoring counter: %v", err)
+	}
+}
+
```

Comment:
Would this make it mandatory to have metrics enabled? I don't think it should be mandatory.

Reply:
We discussed this in DMs. Metrics are always collected, (e.g. `sql_exec_with_metrics`) but they can be published or not, based on the flag `METRICS_ENABLED`.